src/openrct2/world/Park.cpp

```diff
@@ -755,7 +758,7 @@ void Park::UpdateHistories()
     gNumGuestsInParkLastWeek = gNumGuestsInPark;
 
     // Update park rating, guests in park and current cash history
-    HistoryPushRecord<uint8_t, 32>(gParkRatingHistory, CalculateParkRating() / 4);
+    HistoryPushRecord<uint8_t, 32>(gParkRatingHistory, gParkRating / 4);
```

Comment:
This was mainly what caused the issue. It would never use the actual park rating, so it would never record the history correctly for plugins that modify it.